docs/advanced/porting_guide.md

```diff
@@ -37,42 +34,43 @@ Add the target description to ```mbed-os\targets\targets.json```:
+The next step to port a target is to enable the test harness dependencies. To run the test suite, your target must support GPIO, microsecond ticker and serial.
 
 ### GPIO
 
-The ```gpio_s``` is for referencing memory-mapped GPIO registers and passing related pin information or other IO operation data that the HAL needs. These structures are defined in ```objects.h```. The required implementation definition is in ```mbed-os\hal\gpio_api.h```.
-
-### Serial
-
-The ```serial_s``` is for referencing memory-mapped serial registers and passing related pin and peripheral operation information data that the HAL needs. These structures are defined in ```objects.h```. The required implementation definition is in ```mbed-os\hal\serial_api.h```.
+Implement the api declared in `mbed-os/hal/gpio_api.h`. You must define the struct `gpio_t`. This struct is commonly defined in an `objects.h` file within the `mbed-os/targets/TARGET_VENDOR/`, `mbed-os/targets/TARGET_VENDOR/TARGET_MCU_FAMILY` or `mbed-os/targets/TARGET_VENDOR/TARGET_MCU_FAMILY/TARGET_MCUNAME` directories.
 
 ### Microsecond ticker
 
-The microsecond ticker is a system resource that many APIs and other timing utilities use. It needs a one microsecond resolution and should be implemented using a free-running hardware counter or timer with match register. The required implementation definition is in ```mbed-os\hal\us_ticker_api.h```.
+The microsecond ticker is a system resource that many APIs use. It needs a one microsecond resolution and should be implemented using a free-running hardware counter or timer with match register. Implement the api declared in `mbed-os\hal\us_ticker_api.h`.
 
-At this point, we should be able to compile a handful of  tests: 
+At this point, we should be able to compile a handful of tests: 
 
 ``mbed test -m BOARD_NAME --compile -t <toolchain>``
 
-To execute the tests, you'll need to already support [mbed-ls](https://github.com/armmbed/mbed-ls).
+To execute the tests, you need to support [mbed-ls](https://github.com/armmbed/mbed-ls).
+
+### Serial
+
+Implement the api declared in `mbed-os/hal/serial_api.h`. You must define the `serial_t` struct which is used for referencing memory-mapped serial registers and passing related pin and peripheral operation information data that the HAL needs. This struct is defined in `objects.h`.
 
 ### All the others
 
-At this point, the HAL structure should be familiar as a programming model. There are many more APIs to implement, which you enable by adding a ```device_has``` attribute to the MCU_NAME and then providing the implementation. Other HAL APIs include but are not limited to:
+There are many more APIs to implement. You enable the following APIs by adding a `device_has` attribute to the MCU_NAME target definition in `targets.json` and providing an implementation of the api declared in the api header.
 
-```
-device_has       |   api
+device_has       |   api header
```

Comment:
Query: Should these apis be capitalized, or do they need to be lower case to match the code?